### PR TITLE
OTT-1176:  Apply BidAdjustment to imp.BidFloor

### PR DESCRIPTION
**Pull request by Pubmatic-Dhruv-Sonone**
# Description

Please add change description or link to ticket, docs, etc.

# Checklist:

- [ ] PR commit list is unique (rebase/pull with the origin branch to keep master clean).
- [ ] JIRA number is added in the PR title and the commit message.
- [ ] Updated the `header-bidding` repo with appropiate commit id.
- [ ] Documented the new changes.

For Prebid upgrade, refer: https://inside.pubmatic.com:8443/confluence/display/Products/Prebid-server+upgrade

### CODE REVIEW

**Review by pm-nilesh-chate — APPROVED** [review]
Approved with nitpick comments (remove blank lines, spaces and commented code)

File: exchange/utils_ow.go [hunk 1]
```diff
@@ -257,3 +257,19 @@ func createNewContentObject(contentObject *openrtb2.Content, include bool, keys
...
 }
+
+func applyBidAdjustmentToFloor(allBidderRequests []BidderRequest, bidAdjustmentFactors map[string]float64) {
+
+	for _, bidderRequest := range allBidderRequests {
+		bidAdjustment := 1.0
+
+		if bidAdjustemntValue, ok := bidAdjustmentFactors[string(bidderRequest.BidderName)]; ok {
+			bidAdjustment = bidAdjustemntValue
+		}
```
Comment (pm-jaydeep-mohite) [inline exchange/utils_ow.go:268]: Can we skip below code with bidAdjustment is 1.0 

```
if bidAdjustment != 1.0 {
		for index, imp := range bidderRequest.BidRequest.Imp {
 			imp.BidFloor = imp.BidFloor / bidAdjustment
 			bidderRequest.BidRequest.Imp[index] = imp
 		}
}
```

File: exchange/utils_ow_test.go [hunk 2]
```diff
@@ -1047,3 +1047,88 @@ func Benchmark_updateContentObjectForBidder(b *testing.B) {
 		})
 	}
 }
+
+func Test_applyBidAdjustmentToFloor(t *testing.T) {
```
Comment (pm-jaydeep-mohite) [inline exchange/utils_ow_test.go:1051]: To be consistent  rename it to **TestApplyBidAdjustmentToFloor**

File: exchange/utils_ow_test.go [hunk 3]
```diff
@@ -1047,3 +1047,88 @@ func Benchmark_updateContentObjectForBidder(b *testing.B) {
...
+				{
+					BidRequest: &openrtb2.BidRequest{
+						Imp: []openrtb2.Imp{{BidFloor: 100}, {BidFloor: 150}},
+					},
+					BidderName: openrtb_ext.BidderName("appnexus"),
+				},
+			},
+		},
+		{
+			name: "bidAdjustmentFactor present for request bidder",
```
Comment (pm-jaydeep-mohite) [inline exchange/utils_ow_test.go:1106]: Can we have test case like bidAdjustmentFactors present for Pubmatic and bidders present in request Pubmatic & Appnexus ?

File: exchange/utils_test.go [hunk 4]
```diff
@@ -4273,3 +4274,140 @@ func TestGetMediaTypeForBid(t *testing.T) {
...
+					}},
+				},
+				Ext: json.RawMessage(`{"bidder":{"placementId":1}}`),
+			}},
+		},
+	}
+
+	for _, test := range testCases {
+		req := newBidRequest(t)
+		// req.User.Ext = json.RawMessage(`{"consent":"` + test.gdprConsent + `"}`)
```
Comment (pm-jaydeep-mohite) [inline exchange/utils_test.go]: Can we remove commented code?

**Comment by pm-jaydeep-mohite** [comment]
Some code changes were done to support bidAdjustment only for PubMatic adaptor, 
These changes needs to be removed from https://github.com/PubMatic-OpenWrap/prebid-server/blob/7c6706cba2831415efa6a23fceb7005995a9d381/adapters/pubmatic/pubmatic.go#L362

File: exchange/utils_ow.go (hunk 1 above)
Comment (Pubmatic-Dhruv-Sonone) [reply exchange/utils_ow.go:268]: Done

File: exchange/utils_ow_test.go (hunk 2 above)
Comment (Pubmatic-Dhruv-Sonone) [reply exchange/utils_ow_test.go]: Done

File: exchange/utils_ow_test.go (hunk 3 above)
Comment (Pubmatic-Dhruv-Sonone) [reply exchange/utils_ow_test.go:1106]: Done

File: exchange/utils_test.go (hunk 4 above)
Comment (Pubmatic-Dhruv-Sonone) [reply exchange/utils_test.go]: Done

**Comment by Pubmatic-Dhruv-Sonone** [comment]
> Some code changes were done to support bidAdjustment only for PubMatic adaptor, These changes needs to be removed from
> 
> https://github.com/PubMatic-OpenWrap/prebid-server/blob/7c6706cba2831415efa6a23fceb7005995a9d381/adapters/pubmatic/pubmatic.go#L362
> 
> 1. Remove BidAdjustmentFactor member from  reqInfo *adapters.ExtraRequestInfo
> 2. Remove  reqInfo.BidAdjustmentFactor argument from parseImpressionObject()

Done

File: exchange/utils_ow.go [hunk 5]
```diff
@@ -257,3 +257,23 @@ func createNewContentObject(contentObject *openrtb2.Content, include bool, keys
...
+
+		if bidAdjustment == 1.0 {
+			continue
```
Comment (pm-jaydeep-mohite) [inline exchange/utils_ow.go:271]: Can we avoid continue statement ? 

This code would be contributed in prebid server master, and prebid community is against of using continue statement

Comment (Pubmatic-Dhruv-Sonone) [reply exchange/utils_ow.go]: Changed

**Review by pm-jaydeep-mohite — APPROVED** [review]
LGTM

**Pull request merged**